src/lib/yang/schema.lua

```diff
@@ -644,6 +671,8 @@ end
 -- Warn on any "when", resolving them as being true.
 -- Resolve all augment nodes. (TODO)
 function resolve(schema, features)
+   marker.leaf:reset()
+   marker.uses:reset()
```

Comment:
I would rather add the uniqueness detection as a post-pass, once we've parsed everything.  Like the pass that computes whether a node is config or state.  That would avoid the global marker set; the state can be specific to the pass.  WDYT?

Reply:
OK, I was not aware there was already an extra pass. I can add the computation of uniqueness to that pass. Yep, I think in that case the marker won't be necessary as uniqueness state goes appended to each node.